### PR TITLE
Feature/be integ new

### PR DESCRIPTION
**Pull request by erezdrutin**
NOTE: Merging this will break current functionality in Communities page. Same exact integration worked for rides so I guess this is at the component level. I'll try taking a look at some point or we can just merge without it and have someone else take care of it when they have time

### CODE REVIEW

File: apps/ui/src/apis/client.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// Add a response interceptor to log responses
+axiosClient.interceptors.response.use(response => {
+//   console.log('Response:', response);
+  return response;
+}, error => {
+  console.log('Error:', error);
+  return Promise.reject(error);
+});
+
+export default axiosClient;
```
Comment (Kovler10) [inline apps/ui/src/apis/client.ts:23]: please remove irrelevant comments and please avoid using export default

File: apps/ui/src/apis/communities/fetchAllCommunities.ts [hunk 2]
```diff
@@ -1,48 +1,59 @@
...
 
-import { getRandomOption } from '../../utils';
-import { Community } from '@communecar/types';
+interface UserNode {
+  profileImage: string | null;
+}
 
-const options = [tlv, apple, camera];
+interface UserCommunityNode {
+  userByUserId: UserNode;
```
Comment (Kovler10) [inline apps/ui/src/apis/communities/fetchAllCommunities.ts:9]: `UserNode` does not contain user id

File: apps/ui/src/apis/communities/fetchAllCommunities.ts [hunk 3]
```diff
@@ -1,48 +1,59 @@
...
+interface CommunitiesData {
+  allCommunities: {
+    nodes: CommunityNode[];
+  };
+}
 
-const fetchAllCommunities = () => {
-  return communities;
-};
+export const fetchAllCommunities = async (): Promise<Community[]> => {
```
Comment (Kovler10) [inline apps/ui/src/apis/communities/fetchAllCommunities.ts:27]: move exports to end of file

File: apps/ui/src/apis/communities/fetchAllCommunities.ts [hunk 4]
```diff
@@ -1,48 +1,59 @@
...
+            }
+          }
+        }
+      }
+    }`;
+  
+  const data = await graphqlRequest<CommunitiesData>(query);
 
-export { fetchAllCommunities };
+  console.log("communities: ", data);
```
Comment (Kovler10) [inline apps/ui/src/apis/communities/fetchAllCommunities.ts:48]: remove console.log

File: apps/ui/src/apis/communities/fetchAllCommunities.ts [hunk 5]
```diff
@@ -1,48 +1,59 @@
...
+  return data.allCommunities.nodes.map((node): Community => {
+    const pictures = node.userCommunitiesByCommunityId.nodes.map(uc => uc.userByUserId.profileImage).filter((url): url is string => url != null);
```
Comment (Kovler10) [inline apps/ui/src/apis/communities/fetchAllCommunities.ts:50]: please avoid using abbreviations

File: apps/ui/src/apis/rides/fetch-all-rides.ts [hunk 6]
```diff
@@ -1,57 +1,119 @@
...
+  try {
+    const response = await axiosClient.get<LocationResult[]>('/api/v1/external/reverse-geocode', {
+      params: { lat: coords.lat, lon: coords.lon }
+    });
+
+    // Check if the array of results is not empty and return the display name of the first result
+    if (response.data.length > 0 && response.data[0] !== undefined) {
+      return response.data[0].name || response.data[0].displayName;
+    } else {
+      return "מיקום לא ידוע 😵‍💫";
```
Comment (Kovler10) [inline apps/ui/src/apis/rides/fetch-all-rides.ts:108]: change to english

File: apps/BE/src/controllers/external/geocoding/index.ts [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+    if (!location) {
+        return res.status(400).json({ error: "Location parameter is required" });
+    }
+
+    try {
+        const response = await client.get(`/search`, {
+            params: { q: location as string, limit }
+        });
+
+        const results: LocationResult[] = response.data.map((item: any) => ({
```
[the remark this answers is not in the document]
Comment (erezdrutin) [reply apps/BE/src/controllers/external/geocoding/index.ts:73]: There's no point in having this in the types package as this won't be exposed to the FE + adding a type here is extremely redundant IMO but I'll add some BE types folder

File: apps/BE/src/routes/external.route.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+import { getWhatsAppChatUrl, geocodeLocation, reverseGeocodeLocation } from '../controllers/external';
+
+const router = express.Router();
+
+// New route to generate WhatsApp chat URL
+router.get('/whatsapp/:phoneNumber', getWhatsAppChatUrl);
+router.get('/geocode', geocodeLocation);
+router.get('/reverse-geocode', reverseGeocodeLocation);
+
+export default router;
```
[the remark this answers is not in the document]
Comment (erezdrutin) [reply apps/BE/src/routes/external.route.ts:11]: why? this is literally the sole purpose of the file
but sure I'll align with the conventions ya'll used I guess

File: apps/ui/src/apis/communities/fetchAllCommunities.ts (hunk 3 above)
Comment (erezdrutin) [reply apps/ui/src/apis/communities/fetchAllCommunities.ts:27]: we literally have the exact opposite conventions. Hilarious tbh lmfao

File: apps/ui/src/apis/communities/fetchAllCommunities.ts (hunk 2 above)
Comment (erezdrutin) [reply apps/ui/src/apis/communities/fetchAllCommunities.ts:9]: we're not querying it

**Comment by erezdrutin** [comment]
@Kovler10 FYI I aligned FE with your exports requests, keeping BE w/ default exports

**Pull request merged**